### ISSUE TITLE
Cannot add custom data attributes to document field

### ISSUE THREAD

**Issue by mattr** [katalyst/koi#592]
The custom `govuk_document_field` defines a stimulus controller and actions to provide additional functionality over the default file field behaviour; however, we cannot (easily) amend this in a given form to include additional stimulus behaviour.

**Use case**

I want to use javascript to validate the size of a file before it is uploaded.

We can work around this limitation by inheriting from the original document field controller:
```javascript
import KoiDocumentFieldController from "koi/controllers/document_field_controller";

export default class DocumentFieldController extends KoiDocumentFieldController {
  onUpload(e) {
    super.onUpload(e);
    // custom code goes here
  }
}
```
This only allows us to hook into existing actions, and not to define any custom behaviours, however.

Alternatively, we can use the base `govuk_file_field`; however, the trade-off is that we lose the additional functionality that Katalyst has invested in the file upload UX.

**Comment by sfnelson** [comment]
I don't think this request makes sense as-is, especially as Koi controllers are now bundled. For the extension you're describing, I'd consider adding a separate controller and cancelling the upload event, then changing the koi `onUpload` handler to bail out of the event has been cancelled. I don't think that extending this controller itself makes sense however.

FYI the stimulus controller in question has moved to katalyst-govuk-formbuilder. I'm actively working on a feature to move the file upload to JS->activestorage, so the file size validation would happen after upload and before save (server side). I can imagine a pre-upload validation hook there (e.g. via a custom event that can be cancelled) but let's land the new uploaded first.
